File: 02-Prompt-Engineering/6-prompt-enriquecido/typescript-samples/src/examples/0-no-expansion.ts
import { ChatOpenAI } from "@langchain/openai";
import { PromptTemplate } from "@langchain/core/prompts";
import { StringOutputParser } from "@langchain/core/output_parsers";
import {
  loadEnvironment,
  getApiKey,
  getEndpoint,
  printLlmResult,
} from "../utils/helpers";

// Carrega as variáveis de ambiente
loadEnvironment();

async function main(): Promise<void> {
  const llm = new ChatOpenAI({
    model: "gpt-4o",
    temperature: 0.7,
    configuration: {
      baseURL: getEndpoint("GITHUB_MODELS_ENDPOINT"),
      apiKey: getApiKey("GITHUB_MODELS_TOKEN"),
    },
  });

  const prompt = PromptTemplate.fromTemplate(
    "You are a technology assistant. \n" +
      "Answer the following question: \n\n" +
      "{question}"
  );

  const outputParser = new StringOutputParser();

  const chain = prompt.pipe(llm).pipe(outputParser);

  const question = "Explain about the LangChain and LangGraph";

  const answer = await chain.invoke({ question: question });

  console.log(answer);
  console.log(answer.length);
}

main().catch(console.error);

File: 02-Prompt-Engineering/6-prompt-enriquecido/python-samples/examples/0-no-expansion.py
from langchain.prompts import PromptTemplate
from langchain.chat_models import init_chat_model
from langchain_core.output_parsers import StrOutputParser
from dotenv import load_dotenv
from pydantic import SecretStr
import os
import sys

sys.path.append(os.path.join(os.path.dirname(__file__), '..'))

from utils.prompt_helpers import print_llm_result

load_dotenv()

llm = init_chat_model(
    model="gpt-4o",
    temperature=0.7,
    base_url=os.environ["GITHUB_MODELS_ENDPOINT"],
    api_key=SecretStr(os.environ["GITHUB_MODELS_TOKEN"])
)

prompt = PromptTemplate(
  input_variables=["question"],
  template=(
    "You are a technology assistant. \n"
    "Answer the following question: \n\n"
    "{question}"
  ),
)

chain = prompt | llm | StrOutputParser()

question = (
  "Explain about the LangChain and LangGraph"
)

answer = chain.invoke({"question": question})

print(answer)
print(len(answer))